app/src/gui/components/authentication/auth_return.tsx

```diff
@@ -96,6 +99,7 @@ export function AuthReturn() {
 
       // this requires the token
       update_directory();
+      initProjects();
```

Comment:
How does React handle the async ops prior to a navigate - does it assure they are executed? I guess navigate isn't a refresh so the execution should continue in the background?

Reply:
Not really a huge fan of this being added in here - it feels a bit of a hammer solution but that's okay I guess it works.